### PR TITLE
Enable building the Core Tools for integration tests.

### PR DESCRIPTION
**Pull request by Francisco-Gamino**
This PR contains the following:
* Updated the Core Tools build to support building for ingratiation tests. This mode can be set by populating the environment variable `$env:INTEGRATIONBUILDNUMBER`, e.g., `$env:INTEGRATIONBUILDNUMBER = "PreRelease20200904"`. 

  When running in this mode, we download a list of nutget packages that need to be updated from https://raw.githubusercontent.com/Azure/azure-functions-integration-tests/dev/integrationTestsBuild/V3/CoreToolsBuild.json. The nuget packages are pulled from our internal `AzureFunctionsPreRelease` feed and get added via `dotnet add package <packageName> -v <packageVersion> -s <internalFeedName> --no-restore`. After building the Core Tools, we create a `integrationTestBuildManifest.json` file with the build information.

  Here is what the file content looks like:
 ```powershell
{
  "Build": "PreRelease20200903",
  "Packages": {
    "Microsoft.Azure.WebJobs.Script.WebHost": "3.0.14462-ci",
    "Microsoft.Azure.Functions.PowerShellWorker.PS7": "3.0.525"
  },
  "CoreToolsVersion": "3.0.4006"
}
```
* Added `build/scripts/uploadContentToStorageAccount.ps1` to upload the generated Core Tools builds to the storage account for integration tests. When the script is run, it generates the following log:

```powershell
9/3/2020 6:22:18 PM -- Script started.
9/3/2020 6:22:18 PM -- Validating source path 'E:\GH\azure-functions-core-tools\artifacts'.
9/3/2020 6:22:18 PM -- Connecting to storage account...
9/3/2020 6:22:18 PM -- Reading integrationTestBuildManifest.json.
9/3/2020 6:22:18 PM -- Creating version.txt file...
9/3/2020 6:22:47 PM -- Uploading Azure.Functions.Cli.linux-x64.zip...
9/3/2020 6:22:57 PM -- Uploading Azure.Functions.Cli.min.win-x64.zip...
9/3/2020 6:23:12 PM -- Uploading Azure.Functions.Cli.min.win-x86.zip...
9/3/2020 6:23:26 PM -- Uploading Azure.Functions.Cli.no-runtime.zip...
9/3/2020 6:23:41 PM -- Uploading Azure.Functions.Cli.osx-x64.zip...
9/3/2020 6:23:56 PM -- Uploading Azure.Functions.Cli.win-x64.zip...
9/3/2020 6:24:12 PM -- Uploading Azure.Functions.Cli.win-x86.zip...
9/3/2020 6:24:26 PM -- Uploading integrationTestBuildManifest.json...
9/3/2020 6:24:27 PM -- Uploading version.txt...
9/3/2020 6:24:27 PM -- Uploading Azure.Functions.Cli.linux-x64.3.0.4002.zip...
9/3/2020 6:24:43 PM -- Uploading Azure.Functions.Cli.min.win-x64.3.0.4002.zip...
9/3/2020 6:24:59 PM -- Uploading Azure.Functions.Cli.min.win-x86.3.0.4002.zip...
9/3/2020 6:25:13 PM -- Uploading Azure.Functions.Cli.no-runtime.3.0.4002.zip...
9/3/2020 6:25:29 PM -- Uploading Azure.Functions.Cli.osx-x64.3.0.4002.zip...
9/3/2020 6:25:41 PM -- Uploading Azure.Functions.Cli.win-x64.3.0.4002.zip...
9/3/2020 6:25:55 PM -- Uploading Azure.Functions.Cli.win-x86.3.0.4002.zip...
9/3/2020 6:26:10 PM -- Uploading integrationTestBuildManifest.json...
9/3/2020 6:26:10 PM -- Uploading version.txt...
9/3/2020 6:26:10 PM -- Script completed.
```
* Removed unused code `UploadToStorage` in `build/BuildSteps.cs`.

### CODE REVIEW

File: build/BuildSteps.cs
Comment (soninaren) [inline build/BuildSteps.cs:69]: If list fails, restore would fail too. Should skip adding the package incase list fails or and stop the build pipeline if this happens.

**Review by pragnagopa — COMMENTED** [review]
Added minor comments. Please wait for approval from @soninaren  / @ankitkumarr

**Review by soninaren — CHANGES_REQUESTED** [review]
Left some comments. Minor changes. Looks good otherwise. I think you need to rebase the branch

File: build/BuildSteps.cs
Comment (Francisco-Gamino) [reply build/BuildSteps.cs:45]: Fixed.

File: build/BuildSteps.cs
Comment (Francisco-Gamino) [reply build/BuildSteps.cs:49]: Fixed.

File: build/BuildSteps.cs
Comment (Francisco-Gamino) [reply build/BuildSteps.cs:69]: Fixed. Apologies I assumed that `ColoredConsole.Error.WriteLine` would throw automatically or stop the execution. However, I never checked the actual code implementation.

File: build/BuildSteps.cs
Comment (Francisco-Gamino) [reply build/BuildSteps.cs:562]: Fixed. I've added a check in the branch. Here is what it will look like:
`if (!string.IsNullOrEmpty(Settings.IntegrationBuildNumber) && (_manifest != null)) { .. } `

File: build/BuildSteps.cs
Comment (Francisco-Gamino) [reply build/BuildSteps.cs:588]: Fixed -- this is supposed to throw.

File: build/BuildSteps.cs
Comment (Francisco-Gamino) [reply build/BuildSteps.cs:598]: Fixed.

File: build.ps1
Comment (Francisco-Gamino) [reply build.ps1:9]: Fixed.

File: build/BuildSteps.cs
Comment (Francisco-Gamino) [reply build/BuildSteps.cs:46]: Fixed.

File: build/BuildSteps.cs
Comment (Francisco-Gamino) [reply build/BuildSteps.cs:69]: Yes, I agree. I have updated the code, so that if cannot get the package info, we throw and stop the build execution.

File: build.ps1
Comment (soninaren) [reply build.ps1:9]: For some reason i don't see the change in the PR. May be you one of the commit is missing.

**Review by soninaren — APPROVED** [review]

**Review by soninaren — APPROVED** [review]

**Pull request merged**
